src/components/frame/v5/pages/BalancePage/partials/BalanceTable/types.ts

```diff
@@ -8,3 +9,28 @@ export interface BalanceTableFieldModel {
   balance: string | JSX.Element;
   token?: TokenFragment;
 }
+
+export interface TokensDataProps {
+  balance: string;
+  token?: TokenFragment;
+}
+
+export interface TokensProps extends TokensDataProps {
+  isApproved: boolean;
+  isNative: boolean;
+}
+
+export type BalanceTableFilters = {
+  type: {
+    [key: string]: boolean;
+  };
```

Comment:
Storing this as an object feels like overcomplication, why can't it be an array (or a Set) containing token addresses? You can always map that to the same object shape, but we should aim to work with the smallest and simplest form of data.

Reply:
I used this structure because the types require it:

<img width="579" alt="Screen Shot 2023-12-20 at 12 23 38" src="https://github.com/JoinColony/colonyCDapp/assets/119587095/065109a0-4314-48b1-a85a-8c5989ae2dc0">